### ISSUE TITLE
Tag v1.0.0?

### ISSUE THREAD

**Issue by wayneashleyberry** [rubenv/sql-migrate#158]
I know this issue [has been brought up before](https://github.com/rubenv/sql-migrate/issues/122), but I think it's worth reiterating. Adding a `v1.0.0` tag to this project would give a strong signal of stability and maturity to this library. Users would also be able to opt-in to updates more clearly moving forward. GitHub now lets users subscribe to new releases on projects etc. and then changing from `v1.0.0` to `v1.1.0` etc. would be nice and simple.

My go.mod file currently has this line which is not ideal, because according to semver there are no backwards compatibility promises before `v1.0.0`, the next version of `sql-migrate` could completely break my application.

```
github.com/rubenv/sql-migrate v0.0.0-20191116071645-ce2300be8dc8
```

**Comment by rubenv** [comment]
I will never allow any changes that break backwards compatibility.

The annoying thing is that as soon as I tag 1.0.0, I'll have to add a tag for each and every change pushed.

**Comment by wayneashleyberry** [comment]
> each and every change pushed

You don't have to tag every change, you could just tag once a month or every few weeks etc. grouping a few features and fixes together.

There are tools to help automate this and write release notes for you as well, https://github.com/goreleaser/goreleaser is fantastic - once you have your config written it's a single command and you're done.

**Comment by rubenv** [comment]
> you could just tag once a month or every few weeks

That means people won't get bugfixes? How is that better :-)

**Comment by wayneashleyberry** [comment]
as it stands users of this library have no idea what they're getting when they update, and they need to update to a specific commit - to see what's changed you would have to either do a diff or comb through closed pull requests. The tags are just a simple way of grouping changes so that users can see what's been fixed or added etc. I'm more hesitant to update my version of the library now than I would be if it was using [semver](https://semver.org/), if there was a patch update I'd know there are new bugfixes and I could happily update. Tags also let services like [dependabot](https://dependabot.com/) automatically open PR's to update versions in user-land code, meaning we'd probably get the bug fixes sooner than without tags right now.

**Comment by hbroer** [comment]
> I will never allow any changes that break backwards compatibility.

In my opinion this is a other way to say "it is deprecated" :smirk:

**Comment by rubenv** [comment]
It means it's stable.

Why should a good thing that works be deprecated?

**Comment by hbroer** [comment]
Versioning from my perspective is the foundation of a software which is supported over many years. I simply don't use software as a dependency if it does not follow SemVer. There is also something I hear every week from POs, CEOs and other developers: "X will never happen". And it will happen. You say you will never need to break BC but I'm absolute sure you have to in the future (or add a dirty workaround, or make a new repo/module). So it feels deprecated out of the box.

It is just my opinion based on my experience with other projects.

**Comment by rubenv** [comment]
Well go use something else then.

Tip: if you want something done, don't start off by offending the maintainer by calling his project deprecated just because he values stability.

**Comment by hbroer** [comment]
I wasn't offending the maintainer.

**Comment by rubenv** [comment]
You do realize I'm the maintainer, right?

Perhaps this is a cultural thing, but it's hard to classify your comments as constructive.

Now the comments made by @wayneashleyberry, those make sense. Hadn't considered the dependabot angle, so I might tag it for that reason.

But as a friendly advice: remember that most maintainers do this on the side. Be positive and constructive or just stay away. With all the craziness of this quarantine, the last thing I need is someone complaining to get his way (I already have a baby in the house).

**Comment by hbroer** [comment]
I know you are the maintainer, but I did not offend you. You want to read it that way, but that's not my fault :smirk: Maybe I should put in more than one smiley next time ^^ If you feel offended when ever someone says it sounds deprecated out of the box if you don't versioning a package, than you may have no real problems :smirk: :smirk: :smirk:

**Comment by insidieux** [comment]
Hi there

Since you have merged PR for issue #185, may be start making semver tags/releases? 

As was said earlier - this help in some cases:
* using dependabot
* automatically build docker image
* e.t.c.
